src/main/storage/publication-storage.ts

```diff
@@ -15,7 +15,7 @@ import { streamToBufferPromise } from "@r2-utils-js/_utils/stream/BufferUtils";
 import { IZip } from "@r2-utils-js/_utils/zip/zip.d";
 
 import { File } from "readium-desktop/common/models/file";
-import { getFileSize, rmDirSync } from "readium-desktop/utils/fs";
+import { getFileSize, rmDirAsync } from "readium-desktop/utils/fs";
 
```

Comment:
Can you explain the motivation for switching from the `sync` to the `async` `rmDir` variant in the call to `removePublication()`? Was that a *necessary* change in addition to switching from `publicationRepository.delete()` to `catalogService.deletePublication()`? Did you test on Windows, Mac and Linux? (filesystem operations are often inconsistent from one OS to another)

Reply:
Naturally, I understand the impact of "sync" vs. "async" in terms of blocking the Node main thread during the few cycles that are necessary to delete an entire folder (albeit in this particular case: just 3 files, so probably insignificant).

So, here I am asking specifically about fixing the bug: https://github.com/readium/readium-desktop/issues/329
Reading your comment in the issue discussion, it looks like you also came to the conclusion that strictly-speaking, the move from "sync" to "async" was not necessary to fix the bug, and that in fact only the API switch from `publicationRepository.delete()` to `catalogService.deletePublication()` was needed. Is that correct?

Reply:
I removed my works

Reply:
I see, okay. So just to be sure: the bug is fixed *without* the need to switch to async versions of the filesystem functions, right?

Reply:
yes it's works